stickyheaders/src/main/java/com/brandongogetap/stickyheaders/StickyLayoutManager.java

```diff
@@ -133,6 +133,14 @@ public void onAttachedToWindow(RecyclerView view) {
         super.onAttachedToWindow(view);
     }
 
+    @Override
+    public void onDetachedFromWindow(RecyclerView view, RecyclerView.Recycler recycler) {
```

Comment:
This is called after the super call for `RecyclerView#onDetachedFromWindow`. When is the ViewTreeObserver for a view changed when detached? I haven't dug into this recently.

I want to make sure we aren't removing the listener from a different observer than we added it to, which would obviously not solve the potential leak.

Should we be holding onto the observer reference itself in `StickyHeaderPositioner`? We could then re-set it when re-adding the listener.

Reply:
Very good point! I dug in to it and I can tell from sources of api 28 that when the `RecycleView` is detaching, `View#dispatchDetachedFromWindow` is called. Inside `View#dispatchDetachedFromWindow` we have a call to `View#onDetachedFromWindow()` (line 18400 in api 28 source). Inside the `View#onDetachedFromWindow()` our `StickyLayoutManager#onDetachedFromWindow()` is called. This is before the cleaning of the `View#mAttachInfo` (line 18426 in api 28 source) which holds our Listener. So the cleaning is happening to the correct observer (at least at my testing case).

For the last comment, if we should be holding onto the observer itself.. In my opinion we shouldn't, because it would be the source for new memory leaks. If I could, I would get rid the current listener, expose the visibility as a function or in `StickyHeaderHandler` and let the client handle the visibility toggle.

Reply:
Thanks for looking into that. You're right, this seems perfectly safe.